platform/particle_electron/export/aux/uart_bus.hpp

```diff
@@ -0,0 +1,246 @@
+//! \file
+//! \brief Electron UART driver
+#ifndef PLATFORM_UART_BUS_HPP_
+#define PLATFORM_UART_BUS_HPP_
+
+#include <common/bus.hpp>
+#include <ecl/err.hpp>
+#include <ecl/utils.hpp>
+
+#include <cstdint>
+#include <unistd.h>
+
+#include <functional>
+#include <type_traits>
+
+#include <application.h>
+
+namespace ecl
+{
+
+//! Represents distinct peripheral devices
+enum class uart_device
+{
+    serial0,
+    serial1,
+    serial2,
+    serial4,
+    serial5,
+};
+
+//! \brief Electron UART bus wrapper
+template<uart_device dev>
+class uart_bus
+{
+public:
+    // Convenient type aliases.
+    using channel       = ecl::bus_channel;
+    using event         = ecl::bus_event;
+    using handler_fn    = ecl::bus_handler;
+
+    // Static interface only
+    uart_bus() = delete;
+    ~uart_bus() = delete;
+
+    // Should not be copied.
+    uart_bus &operator=(uart_bus&) = delete;
+    uart_bus(uart_bus&) = delete;
```

Comment:
It is just placed there to shut-up warnings from `-Weffc++`. IMO no need to delete all ctors since object cannot be constructed at all (see above).
```
    // Static interface only
    uart_bus() = delete;
    ~uart_bus() = delete;
```